addons/payment_stripe_sca/models/payment.py

```diff
@@ -0,0 +1,275 @@
+        return True
+
+    @api.multi
+    def _registration_render(self, partner_id, qweb_context=None):
+        if qweb_context is None:
+            qweb_context = {}
+        if self.provider == 'stripe':
+            qweb_context.update({'stripe_publishable_key': self.sudo().stripe_publishable_key})
+        return super(PaymentAcquirerStripeSCA, self)._registration_render(partner_id, qweb_context)
+
+
+class PaymentTransactionStripeSCA(models.Model):
+    _inherit = "payment.transaction"
+
+    stripe_payment_intent = fields.Char(string="Stripe Payment Intent ID", readonly=True)
+
+    def _create_stripe_charge(self, acquirer_ref=None, tokenid=None, email=None):
+        raise NotImplementedError("This method can no longer be used with the payment_stripe_sca module.")
+
+    def form_feedback(self, data, acquirer_name):
+        if data.get("reference") and acquirer_name == "stripe":
+            transaction = self.env["payment.transaction"].search([("reference", "=", data["reference"])])
+
+            url = "payment_intents/%s" % transaction.stripe_payment_intent
+            resp = transaction.acquirer_id._stripe_request(url)
+            if resp.get("charges") and resp.get("charges").get("total_count"):
+                resp = resp.get("charges").get("data")[0]
+
+            data.update(resp)
+            _logger.info("Stripe: entering form_feedback with post data %s" % pprint.pformat(data))
+        # note: luckily, the base stripe module did not override this method, avoiding
+        # me from using a context key to avoid this call in the parent model
+        return super(PaymentTransactionStripeSCA, self).form_feedback(data, acquirer_name)
+
+    def _stripe_create_payment_intent(self, acquirer_ref=None, email=None):
+        charge_params = {
+            "amount": int(self.amount if self.currency_id.name in INT_CURRENCIES else float_round(self.amount * 100, 2)),
+            "currency": self.currency_id.name.lower(),
+            "setup_future_usage": "off_session",
+            "confirm": True,
+            "payment_method": self.payment_token_id.stripe_payment_method,
+            "customer": self.payment_token_id.acquirer_ref,
+        }
+        _logger.info("_stripe_create_payment_intent: Sending values to stripe, values:\n%s", pprint.pformat(charge_params))
+
+        res = self.acquirer_id._stripe_request("payment_intents", charge_params)
+        if res.get("charges") and res.get("charges").get("total_count"):
+            res = res.get("charges").get("data")[0]
+
+        _logger.info("_stripe_create_payment_intent: Values received:\n%s", pprint.pformat(res))
+        return res
+
+    def stripe_s2s_do_transaction(self, **kwargs):
+        self.ensure_one()
+        result = self._stripe_create_payment_intent(acquirer_ref=self.payment_token_id.acquirer_ref, email=self.partner_email)
+        return self._stripe_s2s_validate_tree(result)
+
+    def _create_stripe_refund(self):
```

Comment:
not needed in 10 since there was no refund mechanism (refund was introduced with the 'validate card' field which charges then refunds a 1USD tx to validate a card)